File: Codigos en Java/Ejercicio4_Ape06.java
import 
java.util.ArrayList; import 
java.util.Scanner; public 
class Ejercicio4_Ape06 { 
public static void main(String[] args) { 
ArrayList<String> items = new ArrayList<>(); Scanner scan = new 
Scanner(System.in); 
int opcion = 0; 
do { 
System.out.println("\n--- MENU CRUD ---"); 
System.out.println("1. Crear\n2. Leer\n3. Actualizar\n4. Eliminar\n5. Salir"); 
System.out.print("Seleccione: "); 
opcion = scan.nextInt(); 
scan.nextLine(); // Limpiar el buffer 
switch (opcion) { 
case 1: // CREATE 
System.out.print("Ingrese nuevo item: "); 
items.add(scan.nextLine()); // [cite: 26] 
System.out.println("Guardado."); 
break; 
case 2: // READ 
System.out.println("- Lista de Items -"); for (int i = 0; i < items.size(); 
i++) { 
System.out.println("[" + i + "] " + items.get(i)); // [cite: 26] 
}break; 
case 3: // UPDATE 
System.out.print("Indice a actualizar: "); int idxU = 
scan.nextInt(); scan.nextLine(); 
if(idxU >= 0 && idxU < items.size()){ 
System.out.print("Nuevo valor: "); 
items.set(idxU, scan.nextLine()); // set() para actualizar en Java 
System.out.println("Actualizado."); 
} 
break; 
case 4: // DELETE 
System.out.print("Indice a eliminar: "); int idxD = scan.nextInt(); 
if(idxD >= 0 && idxD < items.size()){ 
items.remove(idxD); // [cite: 26] 
System.out.println("Eliminado."); 
} 
break; 
} 
} while (opcion != 5); 
scan.close(); 
} 
}

File: Codigos en C++/Ejercicio4_Ape06.cpp
#include <iostream> 
#include <vector> 
#include <string> 
using namespace std; int 
main() { 
vector<string> items;
int opcion, indice; string 
dato; 
do { 
// Renderizado del Menú 
cout << "\n--- MENU CRUD ---" << endl; 
cout << "1. Crear (Add)\n2. Leer (Read)\n3. Actualizar (Update)\n4. Eliminar (Delete)\n5. Salir" << 
endl; 
cout << "Seleccione: "; cin >> 
opcion; 
switch (opcion) { case 1: // CREATE 
cout << "Ingrese nuevo item: "; cin >> dato; 
items.push_back(dato); // [cite: 26] 
cout << "Guardado." << endl; break; 
case 2: // READ 
cout << "- Lista de Items -" << endl; for (size_t i = 
0; i < items.size(); i++) { 
cout << "[" << i << "] " << items[i] << endl; // [cite: 26] 
} 
break; 
case 3: // UPDATE 
cout << "Indice a actualizar: "; cin >> 
indice; 
if(indice >= 0 && indice < items.size()) { cout <<"Nuevo valor: "; 
cin >> dato; 
items[indice] = dato; // Sobreescritura directa 
cout << "Actualizado." << endl; 
} 
break; 
case 4: // DELETE 
cout << "Indice a eliminar: "; cin >> indice; 
if(indice >= 0 && indice < items.size()) { 
items.erase(items.begin() + indice); // [cite: 26] 
cout << "Eliminado." << endl; 
} 
break; case 5: 
cout << "Saliendo..." << endl; break; 
} 
} while (opcion != 5); 
return 0; 
}